pkg/virt-controller/watch/clone/clone.go

```diff
@@ -345,26 +361,66 @@ func (ctrl *VMCloneController) updateStatus(origClone *clone.VirtualMachineClone
 	return nil
 }
 
-func (ctrl *VMCloneController) createSnapshotFromVm(vmClone *clone.VirtualMachineClone, vm *k6tv1.VirtualMachine, syncInfo syncInfoType) (*snapshotv1.VirtualMachineSnapshot, syncInfoType) {
+func validateVolumeSnapshotStatus(vm *k6tv1.VirtualMachine) error {
+	var vssStatus error
+
+	for _, v := range vm.Spec.Template.Spec.Volumes {
+		if v.PersistentVolumeClaim != nil || v.DataVolume != nil {
+			found := false
+			for _, vss := range vm.Status.VolumeSnapshotStatuses {
+				if v.Name == vss.Name {
+					if !vss.Enabled {
+						vssStatus = errors.Join(vssStatus, fmt.Errorf(ErrVolumeNotSnapshotable, v.Name))
+					}
+					found = true
+					break
+				}
+			}
+			if !found {
+				vssStatus = errors.Join(vssStatus, fmt.Errorf(ErrVolumeSnapshotSupportUnknown, v.Name))
+			}
+		}
+	}
+
+	return vssStatus
+}
+
+func (ctrl *VMCloneController) createSnapshotFromVm(vmClone *clone.VirtualMachineClone, vm *k6tv1.VirtualMachine, syncInfo syncInfoType) syncInfoType {
+	err := validateVolumeSnapshotStatus(vm)
+	if err != nil {
+		return syncInfoType{
+			isClonePending: true,
+			event:          VMVolumeSnapshotsInvalid,
+			reason:         err.Error(),
+		}
+	}
+	if backendstorage.IsBackendStorageNeededForVM(vm) {
+		return syncInfoType{
+			isCloneFailing: true,
+			event:          VMWithBackendStorageInvalid,
+			reason:         ErrVMWithBackendStorage,
+		}
+	}
```

Comment:
A few questions:
1. Should we disallow this only in the cases where the source is a VM? What about snapshot sources? IIUC it is possible to snapshot a VM with a backing storage (keep me honest).
2. WDYT about moving this higher in the stack, maybe under [syncTargetVM()](https://github.com/kubevirt/kubevirt/blob/b537ab91fdf0134d2ecf7e9c760db4db34a42da5/pkg/virt-controller/watch/clone/clone.go#L189-L192)? The problem AFAIK is not in creating the snapshot, but we should fail before even trying to do so.

Reply:
1. regarding snapshot you are right and it was a missing code in the webhook, a PR fixing this was just merged https://github.com/kubevirt/kubevirt/pull/13864
In my other PR https://github.com/kubevirt/kubevirt/pull/13831 I need to rebase on top of this change and handle this also for snapshot source.
2. In next PR Ill move this check to higher up to be checked also for snapshot source